ocrd_network/ocrd_network/processing_worker.py

```diff
@@ -0,0 +1,315 @@
+        except Exception as e:
+            self.log.error(f'Failed to decode processing message body: {body}')
+            self.log.error(f'Nacking processing message with tag: {delivery_tag}')
+            channel.basic_nack(delivery_tag=delivery_tag, multiple=False, requeue=False)
+            raise Exception(f'Failed to decode processing message with tag: {delivery_tag}, reason: {e}')
+
+        try:
+            self.log.info(f'Starting to process the received message: {processing_message}')
+            self.process_message(processing_message=processing_message)
+        except Exception as e:
+            self.log.error(f'Failed to process processing message with tag: {delivery_tag}')
+            self.log.error(f'Nacking processing message with tag: {delivery_tag}')
+            channel.basic_nack(delivery_tag=delivery_tag, multiple=False, requeue=False)
+            raise Exception(f'Failed to process processing message with tag: {delivery_tag}, reason: {e}')
+
+        self.log.info(f'Successfully processed message ')
+        self.log.debug(f'Acking message with tag: {delivery_tag}')
+        channel.basic_ack(delivery_tag=delivery_tag, multiple=False)
+
+    def start_consuming(self) -> None:
+        if self.rmq_consumer:
+            self.log.info(f'Configuring consuming from queue: {self.processor_name}')
+            self.rmq_consumer.configure_consuming(
+                queue_name=self.processor_name,
+                callback_method=self.on_consumed_message
+            )
+            self.log.info(f'Starting consuming from queue: {self.processor_name}')
+            # Starting consuming is a blocking action
+            self.rmq_consumer.start_consuming()
+        else:
+            raise Exception('The RMQConsumer is not connected/configured properly')
+
+    # TODO: Better error handling required to catch exceptions
+    def process_message(self, processing_message: OcrdProcessingMessage) -> None:
+        # Verify that the processor name in the processing message
+        # matches the processor name of the current processing worker
+        if self.processor_name != processing_message.processor_name:
+            raise ValueError(f'Processor name is not matching. Expected: {self.processor_name},'
+                             f'Got: {processing_message.processor_name}')
+
+        # This can be path if invoking `run_processor`
+        # but must be ocrd.Workspace if invoking `run_cli`.
+        workspace_path = processing_message.path_to_mets
+
+        # Build the workspace from the workspace_path
+        workspace = Resolver().workspace_from_url(workspace_path)
+
+        page_id = processing_message.page_id
+        input_file_grps = processing_message.input_file_grps
+        output_file_grps = processing_message.output_file_grps
+        parameter = processing_message.parameters
+        job_id = processing_message.job_id
+
+        # TODO: Find a proper solution for this dirty fix
```

Comment:
Sounds like a general problem for all Tensorflow processors (which there are many). It's nice to patch them all here at once like this, but the TF dependency in core is quite a price to pay. Couldn't we at least make this a conditional import (not even constrained towards a specific processor name)?

Reply:
I wouldn't even make this conditional (on an envvar). I would mandate it. But I would like to spare the effort of adapting all processors. As soon as you are done patching them all, a new one comes along, or TF updates their API and you start over. Hence the single fixture here. But as a conditional import, so you don't need the dependency or any knowledge about the processors in question.

Reply:
Resolved in commit 75b5d4e.